builder/ncloud/step_validate_template.go

```diff
@@ -229,15 +385,80 @@ func (s *StepValidateTemplate) validateServerProductCode() error {
 	return nil
 }
 
+func (s *StepValidateTemplate) validateVpc() error {
+	if !s.Config.SupportVPC {
+		return nil
+	}
+
+	if s.Config.VpcNo != "" {
+		reqParam := &vpc.GetVpcDetailRequest{
+			RegionCode: &s.Config.RegionCode,
+			VpcNo:      &s.Config.VpcNo,
+		}
+
+		resp, err := s.Conn.vpc.V2Api.GetVpcDetail(reqParam)
+		if err != nil {
+			return err
+		}
+
+		if resp == nil || *resp.TotalRows == 0 {
+			return fmt.Errorf("cloud not found VPC `vpc_no` [%s]", s.Config.VpcNo)
+		}
+	}
+
+	if s.Config.SubnetNo != "" {
+		reqParam := &vpc.GetSubnetDetailRequest{
+			RegionCode: &s.Config.RegionCode,
+			SubnetNo:   &s.Config.SubnetNo,
+		}
+
+		resp, err := s.Conn.vpc.V2Api.GetSubnetDetail(reqParam)
+		if err != nil {
+			return err
+		}
+
+		if resp != nil && *resp.TotalRows > 0 && *resp.SubnetList[0].SubnetType.Code != "PUBLIC" {
```

Comment:
Public IP를 생성하려면 Subnet이 Public이어야 하는데요, Subnet(Public)만 입력한 경우 해당 Subnet이 Public인 경우에 `cloud not found public subnet in `vpc_no` 에러를 반환하고 있습니다.

```suggestion
		if resp != nil && *resp.TotalRows > 0 && *resp.SubnetList[0].SubnetType.Code != "PRIVATE" {
```

Reply:
아 등호가 잘못되었네요 감사해요.